app/assets/javascripts/assignment_form.js

```diff
@@ -7,12 +7,13 @@ function importOptions(starterCodeFieldValue) {
   }
 }
 
-function removeErrorBox() {
-  errorBoxes = document.getElementsByClassName("error");
-  for (var i = 0; i < errorBoxes.length; i++) {
-    errorBox = errorBoxes[i]
-    if (errorBox.innerText == "Starter code repository is not a template repository. Make it a template repository to use template cloning.") {
-      errorBox.style.display = 'none';
-    }
-  }
+function removeErrorBox(el) {
+  parent = el.closest(".errored");
+
+  if (!parent) return;
+
+  errorBoxes = parent.getElementsByClassName("error");
+
+  parent.classList.remove("errored");
+  errorBoxes.length && errorBoxes[0].remove();
```

Comment:
If the above change is applied this could be simplified to:

```suggestion
  errorBox && errorBox.remove();
```